modules/apps/journal/journal-web/src/test/java/com/liferay/journal/web/internal/dao/search/JournalRecentArticlesResultRowSplitterTest.java

```diff
@@ -0,0 +1,217 @@
+
+	@ClassRule
+	@Rule
+	public static final LiferayUnitTestRule liferayUnitTestRule =
+		LiferayUnitTestRule.INSTANCE;
+
+	@Before
+	public void setUp() {
+		LanguageUtil languageUtil = new LanguageUtil();
+
+		Mockito.when(
+			_language.get(Mockito.any(Locale.class), Mockito.anyString())
+		).thenAnswer(
+			invocation -> invocation.getArguments()[1]
+		);
+
+		languageUtil.setLanguage(_language);
+	}
+
+	@Test
+	public void testSplit() {
+		ThemeDisplay themeDisplay = _getThemeDisplay();
+
+		LocalDateTime localDateTime = _toLocalDateTime(
+			new Date(), ZoneId.of("UTC"));
+
+		JournalArticle todayJournalArticle = _getJournalArticle(
+			0, localDateTime);
+		JournalArticle oneDayAgoJournalArticle = _getJournalArticle(
+			1, localDateTime);
+		JournalArticle sevenDaysAgoJournalArticle = _getJournalArticle(
+			7, localDateTime);
+		JournalArticle eightDaysAgoJournalArticle = _getJournalArticle(
+			8, localDateTime);
+		JournalArticle thirtyDaysAgoJournalArticle = _getJournalArticle(
+			30, localDateTime);
+		JournalArticle thirtyOneDaysAgoJournalArticle = _getJournalArticle(
+			31, localDateTime);
+
+		List<ResultRow> resultRows = Arrays.asList(
+			_getResultRow(todayJournalArticle),
+			_getResultRow(oneDayAgoJournalArticle),
+			_getResultRow(sevenDaysAgoJournalArticle),
+			_getResultRow(eightDaysAgoJournalArticle),
+			_getResultRow(thirtyDaysAgoJournalArticle),
+			_getResultRow(thirtyOneDaysAgoJournalArticle));
+
+		JournalRecentArticlesResultRowSplitter
+			journalRecentArticlesResultRowSplitter =
+				new JournalRecentArticlesResultRowSplitter(themeDisplay);
+
+		List<ResultRowSplitterEntry> resultRowSplitterEntries =
+			journalRecentArticlesResultRowSplitter.split(resultRows);
+
+		Assert.assertEquals(
+			resultRowSplitterEntries.toString(), 4,
+			resultRowSplitterEntries.size());
+
+		_assertResultRowSplitterEntry(
+			"today", resultRowSplitterEntries.get(0), themeDisplay,
+			todayJournalArticle);
+		_assertResultRowSplitterEntry(
+			"last-7-days", resultRowSplitterEntries.get(1), themeDisplay,
+			oneDayAgoJournalArticle, sevenDaysAgoJournalArticle);
+		_assertResultRowSplitterEntry(
+			"last-30-days", resultRowSplitterEntries.get(2), themeDisplay,
+			eightDaysAgoJournalArticle, thirtyDaysAgoJournalArticle);
+		_assertResultRowSplitterEntry(
+			"older", resultRowSplitterEntries.get(3), themeDisplay,
+			thirtyOneDaysAgoJournalArticle);
+	}
+
+	private void _assertResultRowSplitterEntry(
+		String key, ResultRowSplitterEntry resultRowSplitterEntry,
+		ThemeDisplay themeDisplay, JournalArticle... expectedJournalArticles) {
+
+		Assert.assertEquals(
+			_language.get(themeDisplay.getLocale(), key),
+			resultRowSplitterEntry.getTitle());
+
+		List<ResultRow> resultRows = resultRowSplitterEntry.getResultRows();
+
+		Assert.assertEquals(
+			resultRows.toString(), expectedJournalArticles.length,
+			resultRows.size());
+
+		List<JournalArticle> journalArticles = TransformUtil.transform(
```

Comment:
I think that we don't need this transformation, and we can use this piece of code:

Reply:
for (int i = 0; i < resultRows.size(); i++) {
   ResultRow resultRow = resultRows.get(i);

   Assert.assertEquals(expectedJournalArticles[i], resultRow.getObject());
}